nomos-da/network/subnetworks-assignations/src/versions/history_aware_refill/mod.rs

```diff
@@ -68,12 +88,13 @@ where
     }
 
     fn members_of(&self, network_id: &Self::NetworkId) -> HashSet<Self::Id> {
-        self.assignations
+        let res = self
+            .assignations
             .get(*network_id as usize)
-            .expect("Subntework not found")
-            .iter()
-            .copied()
-            .collect()
+            .map_or_else(HashSet::new, |subnetwork| {
+                subnetwork.iter().copied().collect()
+            });
```

Comment:
```suggestion
            .map_or_else(HashSet::new, HashSet::clone)
```

Reply:
It is actually HashSet of BTreeSets so I think this cannot work like that . I could not make it work with BTreeSet::clone as well.